### PR TITLE
Remove extra space in no_proxy definition

### PR DESCRIPTION
**Pull request by lubronzhan**
### What this PR does / why we need it
Creating management cluster with proxy will fail. Since there are extra space in the no_proxy settings, making the kubeadm command to error out, thus kube-apiserver is not getting correct NO_PROXY value

### Which issue(s) this PR fixes

Fixes #https://github.com/vmware-tanzu/tanzu-framework/issues/3763

### Release note

```release-note
Remove extra space in no_proxy in clusterclass template
```

### CODE REVIEW

**Review by tenczar — CHANGES_REQUESTED** [review]
Can we fix this issue in azure/aws as well.

File: providers/infrastructure-vsphere/v1.3.1/cconly/base.yaml [hunk 1]
```diff
@@ -989,7 +989,7 @@ spec:
               [Service]
               Environment="HTTP_PROXY= {{- .network.proxy.httpProxy -}} "
               Environment="HTTPS_PROXY= {{- .network.proxy.httpsProxy -}} "
-              Environment=" {{- list "localhost" "127.0.0.1" ".svc" ".svc.cluster.local" ((list "IPv6" "DualStack" | has .builtin.cluster.network.ipFamily) | ternary  "::1" nil) | concat .network.proxy.noProxy .builtin.cluster.network.services .builtin.cluster.network.pods | uniq | sortAlpha | join "," -}} "
+              Environment="NO_PROXY= {{- list "localhost" "127.0.0.1" ".svc" ".svc.cluster.local" ((list "IPv6" "DualStack" | has .builtin.cluster.network.ipFamily) | ternary  "::1" nil) | concat .network.proxy.noProxy .builtin.cluster.network.services .builtin.cluster.network.pods | uniq | sortAlpha | join "," -}} "
```
[the remark this answers is not in the document]
Comment (tenczar) [reply providers/infrastructure-vsphere/v1.3.1/cconly/base.yaml:992]: Nevermind. I just double checked, for some reason this error only exists in vsphere.

**Review by tenczar — APPROVED** [review]
LGTM

**Review by HanFa — APPROVED** [review]
lgtm

**Comment by alfredthenarwhal** [comment]
Tests passed! Build no: 3040

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 3048

**Comment by alfredthenarwhal** [comment]
Tests failed! Build no: 3060

**Comment by alfredthenarwhal** [comment]
Build failed! Build no: 3072

**Pull request merged**